src/main/java/auto_scheduler/ScheduleCancelation.java

```diff
@@ -1,13 +1,18 @@
 package org.txstate.auto_scheduler;
 
+import java.io.IOException;
+
 public class ScheduleCancelation implements ScheduleAction {
 	ResourcePathProvider pathProvider;
 
 	public ScheduleCancelation(ResourcePathProvider pathProvider) {
 		this.pathProvider = pathProvider;
 	}
-	public ScheduleSet execute(OperationContext context)
-	{
+	public ScheduleSet execute(OperationContext context) throws IOException {
+		DefaultResourcePathProviderImpl path = new DefaultResourcePathProviderImpl();
+		String studentId = context.getStudentId();
+		StudentScheduleRepositoryImpl schedule = new StudentScheduleRepositoryImpl(path);
+		schedule.delete(studentId);
```

Comment:
as commented on other pull request.

we should not return `null`. instead we should something

```
ScheduleCancelation  extends ScheduletSet {

@override String toString() {
return a string that represents what got successfully canceled.
}
}
```